setup/gcp/installer/util/utils.py

```diff
@@ -325,21 +326,36 @@ def extract_timestamp_from_name(instance_name, include_date=False):
     return instance_name.split('-')[-2][8:]
 
 
-def run_command(cmd_args):
+def run_command(cmd_args, number_of_retry=5, timeout_in_second=10):
     """Wrapper to run a command in subprocess.
 
+    If there is a timeout/error on the API call, we will re try up to 5 times
+    by default.
+
     Args:
         cmd_args (list): The list of command arguments.
+        number_of_retry (int): Number of re try.
+        timeout_in_second (int): Timeout in second.
 
     Returns:
         int: The return code. 0 is "ok", anything else is "error".
         str: Output, if command was successful.
         err: Error output, if there was an error.
     """
+
     proc = subprocess.Popen(cmd_args,
                             stdout=subprocess.PIPE,
                             stderr=subprocess.PIPE)
+
+    timer = threading.Timer(timeout_in_second, proc.kill)
+
+    timer.start()
     out, err = proc.communicate()
+    timer.cancel()
+
+    if proc.returncode and number_of_retry > 0:
+        return run_command(cmd_args, number_of_retry - 1)
```

Comment:
Is there a message to tell the users what retry number this is?  And a message that there is an error and retry is necessary?

And exponential retry?  What do you think?

Reply:
What about printing a message like 'There is an error executing command CMD_ARGS, retrying...' ? Do you think the user needs to know/ cares about the current number retry?

Reply:
Do you think exponential retry can be sometimes too long? I think if the command takes more than 10 seconds to run then there might be some issues on that already.